syz-manager/html.go

```diff
@@ -334,42 +334,24 @@ func (mgr *Manager) httpCoverCover(w http.ResponseWriter, r *http.Request, funcF
 		coverFilter = mgr.coverFilter
 	}
 
-	if funcFlag == DoRawCoverFiles {
-		if err := rg.DoRawCoverFiles(w, progs, coverFilter); err != nil {
-			http.Error(w, fmt.Sprintf("failed to generate coverage profile: %v", err), http.StatusInternalServerError)
-			return
-		}
-		runtime.GC()
-		return
-	} else if funcFlag == DoRawCover {
-		rg.DoRawCover(w, progs, coverFilter)
-		return
-	} else if funcFlag == DoFilterPCs {
-		rg.DoFilterPCs(w, progs, coverFilter)
-		return
-	}
-
-	do := rg.DoHTML
-
-	if funcFlag == DoHTMLTable {
-		do = rg.DoHTMLTable
-	} else if funcFlag == DoModuleCover {
-		do = rg.DoModuleCover
-	} else if funcFlag == DoCSV {
-		do = rg.DoCSV
-	} else if funcFlag == DoCSVFiles {
-		do = rg.DoCSVFiles
-	}
-
-	debug := r.FormValue("debug") != ""
-
 	params := cover.CoverHandlerParams{
 		Progs:       progs,
 		CoverFilter: coverFilter,
-		Debug:       debug,
+		Debug:       r.FormValue("debug") != "",
+	}
+
+	flagToFunc := map[int]func(w http.ResponseWriter, params cover.CoverHandlerParams) error{
```

Comment:
Maybe drop the map then and just pass the right callback to httpCoverCover()?

Reply:
Not easy. Check the update code.
tools/syz-cover changed the reality.